src/pages/index.js

```diff
@@ -54,26 +124,77 @@ addButton.addEventListener("click", () => {
   addNewCardFormValidator.disableSubmitButton();
   addNewCardModal.open();
 });
+updateAvatarIcon.addEventListener("click", () => {
+  updateAvatarFormValidator.disableSubmitButton();
+  updateAvatarModal.open();
+});
 
 export const userInfo = new UserInfo({
   userNameSelector: ".profile__heading",
   userOccupationSelector: ".profile__sub-heading",
+  profileAvatarSelector: ".profile__avatar",
 });
 
 export function updateProfile(profileObject) {
   const newName = profileObject["input-name"];
   const newOccupation = profileObject["input-description"];
+  console.log(profileObject);
+  profileSubmitButton.textContent = "Saving...";
+
+  api
+    .updateProfile({
+      name: newName,
+      occupation: newOccupation,
+    })
+    .finally(() => {
+      profileSubmitButton.textContent = "Save";
+    });
   userInfo.setUserInfo(newName, newOccupation);
```

Comment:
```js

  userInfo.setUserInfo(newName, newOccupation);
  editProfileFormValidator.disableSubmitButton();
  editProfileModal.close();
```

This should be inside `then`


You need to update `DOM` only in blocks `then` after a successful response from the server so that when there is an error it wouldn’t change


**Please, correct it everywhere**